### ISSUE TITLE
New app request/suggestion: OpenProject 

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1229]
Shantanu via the [forums](https://www.turnkeylinux.org/comment/30414#comment-30414) has suggested/requested an [OpenProject](https://www.openproject.org/) server (the OP of that thread, kaktux was also seeking assistance to install it, so I guess they would also be a supporter). Shantanu goes on to say:

> OpenProject (vs. Redmine) [has] Slicker GUI, bigger fonts, automatic repo creation (which really works in OpenProject) [...]

It does look really slick and as a fork of Redmine, should hopefully be fairly easy to create (i.e. start with Redmine appliance and swap out Redmine for OpenProject).

**Comment by MPTMG** [comment]
Do you want me to work on this?

**Comment by JedMeister** [comment]
FWIW I've just had a community member ask about this.

He had a go at setting it up himself and couldn't get it running. Initially he tried with the LAPP appliance as the base, then after that failed, retried on a Core base but still no joy...

**Comment by MPTMG** [comment]
FYI, tkl-dev is failing to build for me after a fresh install just building "core" with no changes. Below is the error. It is throwing an error looking for a package " jitterentropy-rngd" that is no longer available. Perhaps there is a bug in tkl-dev. I installed a new tkl-dev VM today to test. As always correct me if I am wrong and you are able to build without issue, but presently I cannot build anything with tkl-dev.

Warning: Deferring update-initramfs -c -t -k 4.9.0-11-amd64 -b /boot
Setting up linux-image-amd64 (4.9+80+deb9u9) ...
W: --force-yes is deprecated, use one of the options starting with --allow instead.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package jitterentropy-rngd
Traceback (most recent call last):
  File "/usr/lib/fab/wrapper.py", line 34, in <module>
    CliWrapper.main()
  File "/usr/lib/python2.7/dist-packages/pyproject.py", line 388, in main
    commands.run(name, args)
  File "/usr/lib/python2.7/dist-packages/pyproject.py", line 261, in run
    command.module.main()
  File "/usr/lib/fab/pylib/cmd_install.py", line 129, in main
    installer.install(packages, ignore_errors)
  File "/usr/lib/fab/pylib/installer.py", line 267, in install
    self._install(packages, ignore_errors)
  File "/usr/lib/fab/pylib/installer.py", line 137, in _install
    self.chroot.system("apt-get", *(args + packages))
  File "/usr/lib/python2.7/dist-packages/chroot.py", line 80, in system
    executil.system(*self._prepare_command(*command))
  File "/usr/lib/python2.7/dist-packages/executil.py", line 56, in system
    raise ExecError(command, exitcode)
executil.ExecError: non-zero exitcode (100) for command: chroot '/turnkey/fab/products/core/build/root.build'  'sh'  '-c'  "env -i  'TERM=xterm-256color'  'AMD64=y'  'DEBIAN_FRONTEND=noninteractive'  'HOSTNAME=core'  'WEBMIN_FW_TCP_INCOMING=22 80 443 12320 12321'  'FAB_ARCH=amd64'  'DEBIAN=y'  'FAB_HTTP_PROXY=http://127.0.0.1:8124'  'DEBIAN_PRIORITY=critical'  'HOME=/root'  'RELEASE=debian/stretch'  'PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/bin:/usr/sbin'  'CODENAME=stretch'  'LC_ALL=C'  'DISTRO=debian' apt-get 'install'  '--force-yes'  '--assume-yes'  'libgcc1'  'libnet-ssleay-perl'  'lsb-base'  'kbd'  'webmin-filemin'  'findutils'  'rsyslog'  'sensible-utils'  'systemd'  'util-linux'  'libncursesw5'  'dtach'  'whiptail'  'less'  'sysv-rc'  'shellinabox'  'dpkg'  'iptables'  'libpam-modules'  'liblzma5'  'initramfs-tools-core'  'cpio'  'bash-completion'  'libacl1'  'debian-archive-keyring'  'startpar'  'libncurses5'  'grep'  'libgpg-error0'  'acpi-support-base'  'libpam0g'  'bind9-host'  'webmin-updown'  'stunnel4'  'libpam-modules-bin'  'webmin'  'psmisc'  'e2fslibs'  'tklbam'  'net-tools'  'hubdns'  'systemd-sysv'  'udhcpc'  'libfile-mimeinfo-perl'  'debconf'  'webmin-authentic-theme'  'sysvinit-utils'  'webmin-sshd'  'adduser'  'syslinux-common'  'webmin-passwd'  'udev'  'cron'  'sed'  'live-tools'  'mktemp'  'kmod'  'rsync'  'localepurge'  'nano'  'isolinux'  'python3-bottle'  'libselinux1'  'dehydrated'  'passwd'  'jitterentropy-rngd'  'libsmartcols1'  'webmin-net'  'libpam-runtime'  'libss2'  'live-boot-initramfs-tools'  'ntp'  'screen'  'libpcre3'  'libdebconfclient0'  'libmount1'  'libattr1'  'webmin-useradmin'  'wget'  'libbz2-1.0'  'python3-requests'  'bash'  'libsystemd0'  'mawk'  'di-live'  'webmin-software'  'login'  'libuuid1'  'linux-base'  'libklibc'  'etckeeper'  'authbind'  'python3-dialog'  'gpgv'  'apt'  'lsb-release'  'init-system-helpers'  'libc-bin'  'syslinux'  'tar'  'perl-openssl-defaults'  'liblz4-1'  'cron-apt'  'webmin-syslog'  'netbase'  'libaudit-common'  'webmin-raid'  'turnkey-sysinfo'  'webmin-custom'  'klibc-utils'  'libsepol1'  'traceroute'  'gzip'  'libudev1'  'bsd-mailx'  'libgcrypt20'  'hdparm'  'debianutils'  'bsdutils'  'webmin-tklbam'  'git'  'ncurses-base'  'confconsole'  'iputils-ping'  'webmin-fdisk'  'eject'  'mount'  'lvm2'  'initscripts'  'resolvconf'  'libdb5.3'  'gnupg'  'libapt-pkg5.0'  'procps'  'ethtool'  'libprocps6'  'live-boot'  'man-db'  'libsemanage-common'  'dash'  'logrotate'  'unzip'  'fail2ban'  'file'  'base-files'  'curl'  'initramfs-tools'  'libtinfo5'  'libfdisk1'  'hostname'  'libaudit1'  'coreutils'  'libsemanage1'  'libkmod2'  'webmin-fail2ban'  'webmin-shell'  'libc-l10n'  'insserv'  'ncurses-bin'  'gcc-6-base'  'libcap-ng0'  'multiarch-support'  'iproute2'  'zlib1g'  'libc6'  'webmin-firewall'  'libustr-1.0-1'  'libstdc++6'  'libcomerr2'  'postfix'  'diffutils'  'ncurses-term'  'inithooks'  'e2fsprogs'  'base-passwd'  'perl-base'  'locales'  'ifupdown'  'vim-tiny'  'webmin-lvm'  'libblkid1'  'webmin-postfix'  'ssh'"
/usr/share/fab/product.mk:535: recipe for target 'build/stamps/root.build' failed
make: *** [build/stamps/root.build] Error 1

Additionally, I was checking to see if it was possible to simply install jitterentropy-rngd, but it fails to find the package in "Debian 9".

**Comment by JedMeister** [comment]
Really sorry about that @MPTMG - that's all my fault. :cry: 

**TL;DR**

if you checkout the `15.x` branch of common you should be able to build v15.x appliances still. I.e.:

```
cd common
git checkout 15.x
```
OTTOMH that should be enough to build v15.x appliances again. Although alternatively, you may prefer to wait a few days or so? Fingers crossed it won't be much longer before I upload packages, at least to `buster-testing`.

----

**Long version**

I jumped the gun a bit merging stuff into master for v16.0. We have v16.0 builds working on TKLDev v15.0, but it's quite a convoluted process to set up. It's sort of documented in the TKLDev docs although it's not clearly laid out.... They need a really good tidy up!

FWIW, some relevant stuff is in the TKLDev [advanced docs](https://github.com/turnkeylinux-apps/tkldev/tree/master/docs/advanced). The [transition docs](https://github.com/turnkeylinux-apps/tkldev/tree/master/docs/transition) should be somewhat relevant too, but hasn't been updated for ages. There is quite possibly also some relevant stuff in [development docs](https://github.com/turnkeylinux-apps/tkldev/tree/master/docs/development) as well...

Regardless, it's not very clear and there are likely a lot of bugs, mistakes and omissions, some due to lack of maintenance, some due to curse of knowledge... Besides even if the docs were right, the process is still quite convoluted and time consuming. FYI, you need to download all the required Debian packages, build a bootstrap then a buildroot, then all the TurnKey packages, before you can even start with an appliance... If you're interested, then let me know and perhaps I could put aside a little time to give you some clearer pointers?

I had hoped that by now we would have had all the v16.x packages uploaded to our apt repo server, so I could just tell you to set `RELEASE=debian/buster` and you'd be able to build v16.0 pre-release appliances. But unfortunately, that's not the case yet... FWIW I've been somewhat diverted to upgrading some of our backend infrastructure and processes which is in turn holding up uploading the new Buster packages... I should have them up soon as I really want to get the Core v16.0RC1 published.

**Comment by MPTMG** [comment]
I suffer from the curse of ignorance. I haven't kept pace with the changes to the docs. Super impressed that as of 14.2 I can build the custom release types. Thanks for pointing me to the updated docs and providing the cure for ignorance. FWIW I appreciate all the hard work you guys put into TKL and I hope to assist in expanding the library of appliances in the coming year.

**Comment by JedMeister** [comment]
Thanks for you kind words @MPTMG. The docs are far from perfect and likely some bits are well out of date, but I guess it's better than nothing...

If you have any questions or anything needs clarification, then please ask. If you spot any glaring errors or omissions, please feel free to open a pull request.

You've provided some fantastic apps to date and we've really appreciated it. I look forward to see what more you come up with! :smile:
